Mj review (#479)
worldbank/dime-data-handbook: chapters/2-collaboration.tex

Therefore, it is a useful tool for managing non-code-related tasks.

% ----------------------------------------------------------------------------------------------
\subsection{Setting up a productive code environment}

Taking time early in your project to choose a programming language for your project and

> setting up a productive code environments for that language

Reviewer: ```suggestion
setting up productive code environments for that language
```

will make your work significantly easier.
  \index{code environments}
Setting up a productive code environment means that 
you make sure that the programming language and all other software your code requires
will run smoothly on all the computers (laptop, servers etc.) 
that you need it to run on.
It also means that you have a productive way of interacting with your code,

> and that the code has an seamless way of accessing the data.

Reviewer: ```suggestion
and that the code has a seamless method to access your data.
```

While this means you will inevitably have different projects
with different code environments, each one will be better than the last,
and you will avoid the costly process of migrating an ongoing project

> into a new code environment.
> Code environments should be documented as precise as possible.
> The specific version number of the programming languages and the individual packages you use

Reviewer: ```suggestion
into a new code environment.
Code environments should be documented as precisely as possible.
The specific version number of the programming languages and the individual packages you use
```

should be referenced or maintained so that they can be reproduced going forward,
even if different releases contain changes that would break your code
or change your results.
  \index{software versions}
DIME Analytics developed the command \texttt{ieboilstart} in the \texttt{ietoolkit} package
to support version and settings stability in Stata.\sidenote{
  \url{https://dimewiki.worldbank.org/ieboilstart}}

> If your project requires more than one programming language,
> for example if you analyze your data in one language but visualize your results in another,
> then make sure to make an as clear division between the two as possible. 
> This means that you first complete all tasks done in one language, 
> before the completing the rest of the tasks in the other language.
> Frequently swapping back and forth between languages is a reproducibility nightmare.

Reviewer: ```suggestion
If your project uses more than one programming language,
such as analyzing data in one language but visualizing results in another,
make as clear of a division between the two as possible and document how to execute both. Frequently swapping back and forth between languages is a reproducibility nightmare
if it is not written in a fully stable and self-executing way,
because manually switching execution environments is very burdensome on others.
```

which gives you directory-level views and file management actions,
such as folder management, Git integration,
and simultaneous work with other types of files, without leaving the editor.
Using an external editor can also be preferable since your editor will not crash

> if the execution of your code makes Stata or R to crash.

Reviewer: ```suggestion
if the execution of your code causes you statistical software to crash.
```
